src/develop/pixelpipe_hb.c

```diff
@@ -2243,6 +2245,8 @@ int dt_dev_pixelpipe_process(dt_dev_pixelpipe_t *pipe, dt_develop_t *dev, int x,
   if(pipe->devid >= 0) dt_opencl_events_reset(pipe->devid);
 
   dt_iop_roi_t roi = (dt_iop_roi_t){ x, y, width, height, scale };
+  pipe->final_width = width;
+  pipe->final_height = height;
```

Comment:
A bit later in this routine we have:

```
  // terminate
  dt_pthread_mutex_lock(&pipe->backbuf_mutex);
  pipe->backbuf_hash = dt_dev_pixelpipe_cache_hash(pipe->image.id, &roi, pipe, 0);
  pipe->backbuf = buf;
  pipe->backbuf_width = width;
  pipe->backbuf_height = height;
```

So to me backbuf_width/height is final_width/height, no?

Reply:
Unfortunately not, further down while processing the pipeline this holds dimensions of a valid display buffer, this is zero while exporting to file/thumbnail.

just pushed commit displays data via -d roi while processing the module...